Frameworks/Foundation/NSOperation.mm

```diff
@@ -344,14 +347,14 @@ - (NSArray*)dependencies {
  @Status Interoperable
 */
 - (void)dealloc {
-    { // _dependenciesLock scope
-        std::lock_guard<std::recursive_mutex> lock(_dependenciesLock);
```

Comment:
It wasn't ever. Shouldn't need to lock anything in dealloc, by which time no one should be referencing the object anymore.